src/witty/compile_module.py

```diff
@@ -114,6 +337,41 @@ def compile_module(
     ModuleType
         The compiled module.
     """
+
+    return _compile_module(
+        source_pyx,
+        source_type="pyx",
+        source_files=source_files,
+        include_dirs=include_dirs,
+        library_dirs=library_dirs,
+        language=language,
+        extra_compile_args=extra_compile_args,
+        extra_link_args=extra_link_args,
+        name=name,
+        force_rebuild=force_rebuild,
+        quiet=quiet,
+        output_dir=output_dir,
+        **extension_kwargs,
+    )
+
+
+def _compile_module(
+    source: str,
+    source_type: Literal["pyx", "nanobind"],
+    *,
+    source_files: Sequence[Path | str] = (),
+    include_dirs: Sequence[Path | str] = (".",),
+    library_dirs: Sequence[Path | str] = (),
+    language: Literal["c", "c++"] | None = None,
+    extra_compile_args: list[str] | None = None,
+    extra_link_args: list[str] | None = None,
+    build_extra_objects: callable[[], str] | None = None,
+    name: str = None,
```

Comment:
```suggestion
    build_extra_objects: Callable[[Path], list[str]] | None = None,
    name: str | None = None,
```

unfortunately, the function `callable` isn't valid as a type hint, and signature needed updating